src/Tizen.Network.Stc/Tizen.Network.Stc/StcManager.cs

```diff
@@ -0,0 +1,109 @@
+            this.SetHandle(IntPtr.Zero);
+            return true;
+        }
+    }
+
+    /// <summary>
+    /// A class which is used to manage Smart Traffic control (Stc).<br/>
+    /// </summary>
+    public static class StcManager
+    {
+        /// <summary>
+        /// Gets the Stc safe handle.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <returns>The instance of the SafeStcHandle.</returns>
+        /// <privilege>http://tizen.org/privilege/network.get</privilege>
+        /// <exception cref="NotSupportedException">Thrown when the Stc is not supported.</exception>
+        /// <exception cref="UnauthorizedAccessException">Thrown when the permission is denied.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when the method failed due to an invalid operation.</exception>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        internal static SafeStcHandle GetStcHandle()
+        {
+            return StcManagerImpl.Instance.GetSafeHandle();
+        }
+
+        /// <summary>
+        /// Gets the statistics information of an application, which matched the rule, asynchronously. The rule must contain an Application ID.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <param name="rule"> The StcRule object.</param>
+        /// <returns>The Statistics information of the application that used network in between timestamps specified.</returns>
+        /// <privilege>http://tizen.org/privilege/network.get</privilege>
+        /// <exception cref="NotSupportedException">Thrown when the Stc is not supported.</exception>
+        /// <exception cref="UnauthorizedAccessException">Thrown when the permission is denied.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when the method failed due to an invalid operation.</exception>
+        public static Task<NetworkStatistics> GetStatisticsAsync(StcRule rule)
+        {
+            return StcManagerImpl.Instance.GetStatisticsAsync(rule);
+        }
+
+        /// <summary>
+        /// Gets statistics information of each application that used network in between specified timestamps and matches the given rule, asynchronously.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <param name="rule"> The StcRule object.</param>
+        /// <returns>A list of the NetworkStatistics objects.</returns>
+        /// <privilege>http://tizen.org/privilege/network.get</privilege>
+        /// <exception cref="NotSupportedException">Thrown when the Stc is not supported.</exception>
+        /// <exception cref="UnauthorizedAccessException">Thrown when the permission is denied.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when the method failed due to an invalid operation.</exception>
+        public static Task<IEnumerable<NetworkStatistics>> GetAllStatisticsAsync(StcRule rule)
```

Comment:
Which values are used to filter in StcRule?
Is the appid specified in the StcRule ignored?

Reply:
appid is used in "GetStatisticsAsync", while GetAllStatisticsAsync does not include appid, as it gives stats for every app.

Reply:
So, You need to remove appid property in StcRule data.
This API design is easy to make mistakes and easy to misreading API usage

Reply:
typedef struct {
    char app_id[STC_APP_ID_LEN];
    stc_time_interval_s interval;
    stc_iface_type_e iface_type;
    stc_time_period_e time_period;
 } stc_stats_rule_s;

To get statistics, filtered on the basis of different parameters(combinations), we set corresponding elements of StcRule and use appropriate API to get Statistics. This is how CAPI is designed, and also CAPI recently had changes related to these.
GetStatisticsAsync() filters on the basis of appId(and time interval) provided by given StcRule. So we needed AppId setter and getter in StcRule.

Reply:
On `GetStatisticsAsync` method, AppId can be passed as a parameter like below
``` c#
 public static Task<NetworkStatistics> GetStatisticsAsync(string appId, StcRule rule)
```

Reply:
There is no need to send 'appId' separately, It could be set in the above rule parameter only.

Reply:
Why do you think so?
I recommend to remvoe appid property from StcRule class. because it has no means some context(GetAllStatisticsAsync).
if appid is need in GetStatisticsAsync, we can pass as separatly parameter

Reply:
I have made changes as per suggestion.